### ISSUE TITLE
`Naming/ClassAndModuleCamelCase` incorrectly suggests writing class/module name `X86_64` in CamelCase

### ISSUE THREAD

**Issue by postmodern** [rubocop/rubocop#11611]
I have a module namespace called `X86_64` which represents the x86-64 architecture. `rubocop` seems to think this module/class name should be written in CamelCase. This is of course impossible, because numeric digits cannot be uppercase nor lowercase. Grouping the digits together (ex: `X8664`) would make the class name unrecognizable. `rubocop` should allow for class/module names that contain multiple groupings of digits separated by an underscore (ex: `X86_64`, `ISO_8859_1`, `CVE_2023_1234`, etc).

## Expected behavior

No warnings.

## Actual behavior

```
For /home/postmodern/test/ruby/rubocop: configuration from /home/postmodern/test/ruby/rubocop/.rubocop.yml
Default configuration from /home/postmodern/test/ruby/rubocop/vendor/bundle/ruby/3.1.0/gems/rubocop-1.45.1/config/default.yml
Use parallel by default.
Running parallel inspection
Loading cache from /home/postmodern/.cache/rubocop_cache/da11a5156d303827c8892095e0ccf80789af8f4c/6d7a3b621ca1730e04accd938619e4bdab66cfb1/16b1d2cd5a129139444f764f3f107677640b95f9
Inspecting 2 files
Scanning /home/postmodern/test/ruby/rubocop/Gemfile
Loading cache from /home/postmodern/.cache/rubocop_cache/da11a5156d303827c8892095e0ccf80789af8f4c/6d7a3b621ca1730e04accd938619e4bdab66cfb1/16b1d2cd5a129139444f764f3f107677640b95f9
CScanning /home/postmodern/test/ruby/rubocop/test.rb
Loading cache from /home/postmodern/.cache/rubocop_cache/da11a5156d303827c8892095e0ccf80789af8f4c/6d7a3b621ca1730e04accd938619e4bdab66cfb1/84b2471cdd32b8591f6b01d71470e62a6eb625b3
C

Offenses:

Gemfile:1:1: C: [Correctable] Style/FrozenStringLiteralComment: Missing frozen string literal comment.
source 'https://rubygems.org'
^
test.rb:1:1: C: [Correctable] Style/FrozenStringLiteralComment: Missing frozen string literal comment.
#
^
test.rb:4:8: C: Naming/ClassAndModuleCamelCase: Use CamelCase for classes and modules.
module X86_64
       ^^^^^^

2 files inspected, 3 offenses detected, 2 offenses autocorrectable
Finished in 0.46626606299969353 seconds
```

## Steps to reproduce the problem

`Gemfile`:

```ruby
source 'https://rubygems.org'

gem 'rubocop'
```

`test.rb`:

```ruby
#
## The x86-64 architecture module.
#
module X86_64
end
```

1. `bundle install`
2. `bundle exec rubocop --debug`

## RuboCop version

```
$ rubocop -V
1.45.1 (using Parser 3.2.1.0, rubocop-ast 1.26.0, running on ruby 3.1.3) [x86_64-linux]
```

**Comment by philliplongman** [comment]
In the meantime, you can add it to your `.rubocop.yml`
```yaml
Naming/ClassAndModuleCamelCase:
  AllowedNames:
    - parent_module
    - X86_64
```

**Comment by bbatsov** [comment]
I'm not sure this is something we need to "fix", given that the configuration exists exactly for such edge cases that appear pretty rarely in practice.

**Comment by postmodern** [comment]
@bbatsov that logic could be applied to all rubocop false-positive issues? I think the regex for class/module names could be expanded to something like `/([A-Z][a-z0-9]*)+(?:_\d+)*/`. Alternatively, rubocop could contain a list of known class/module names that defy CamelCase due to containing multiple numbers (ex: `X86_64`, `ISO_8859_1`, etc).

**Comment by bbatsov** [comment]
Or this can just be some "AllowedPattern" that people can add to their config. I get your point, but I still believe those names are so rare that they don't need a general solution and so be handled on a case-by-case basis. Perhaps that's also an opportunity to improve the documentation of the cop.
